### PR TITLE
fix(files-pane): wire up git change indicators (#611)

### LINKED ISSUE #611: Files pane: wire up git change indicators for file icons
## Context

The `Session.changedFiles` field (`FileArtifact[]`) exists on the session interface but is never populated with actual git status data. It's initialized as `[]` in every session creation path (`useSessionCrud.ts`, `useWizardHandlers.ts`, `worktreeSession.ts`, `tabHelpers.ts`) and no code ever writes to it.

The file explorer already has plumbing to consume this data:
- `FileExplorerPanel.tsx:941` — `const change = session.changedFiles?.find((f) => f.path.includes(node.name));`
- `FileExplorerPanel.tsx:1021` — passes `change?.type` to `getExplorerFileIcon()`
- Default theme has `fileTypeColor()` that maps `added`→green, `modified`→yellow, `deleted`→red

But since `changedFiles` is always empty, none of this has any visible effect.

Additionally, PR #609 (Files Pane Icon Theme) adds a "Rich" icon theme that accepts but ignores the `_type` parameter entirely in `getRichExplorerFileIcon()`. Once `changedFiles` is wired up, the rich theme will need to handle diff status too (e.g., colored dot overlay, border, or CSS filter on the `` SVG icons).

## What needs to happen

1. **Populate `changedFiles`** — Run `git status` (or equivalent) and write results into `session.changedFiles` with proper `FileArtifact.type` values (`added`, `modified`, `deleted`). This should refresh on file tree refresh and after agent operations complete.

2. **Default theme** — Already wired via `fileTypeColor()`. Should work once data flows. Verify visually.

3. **Rich theme** — `getRichExplorerFileIcon` currently discards `_type`. Add a visual indicator for change status — options include:
   - Colored dot/badge overlay on the icon
   - CSS filter or opacity shift
   - Colored left border on the file row
   
   Whatever approach is chosen, it should be consistent with how the default theme communicates the same information.

4. **File name match** — The current lookup (`f.path.includes(node.name)`) is fragile and could false-match (e.g., a file named `utils.ts` matching `src/other/utils.ts`). Consider matching against the full relative path instead.

## Relevant files

- `src/renderer/types/index.ts:531` — `changedFiles: FileArtifact[]` definition
- `src/renderer/components/FileExplorerPanel.tsx:941` — consumption site
- `src/renderer/utils/fileExplorerIcons/richTheme.tsx` — `getRichExplorerFileIcon` (ignores `_type`)
- `src/renderer/utils/fileExplorerIcons/defaultTheme.tsx` — `getDefaultExplorerFileIcon` (has `fileTypeColor`)
- `src/renderer/utils/fileExplorerIcons/shared.ts` — shared types including `FileChangeType`

### PR DESCRIPTION
**Pull request by chr1syy**
## Summary

- The Files pane has been silently broken because `session.changedFiles` is never populated; the file-tree never showed added/modified/deleted indicators. Switch the row renderer to read from `useGitDetail()` (already polled per session by `GitStatusProvider`).
- Drop the fragile `f.path.includes(node.name)` lookup in favour of a full-relative-path match. The old version false-matched siblings sharing a leaf name (e.g. `package.json` lighting up because `src/package-loader.ts` was modified).
- Walk the tree to highlight ancestor folders that contain changed descendants (VS Code style, per the #611 follow-up comment).
- Replace the tinted icon + uppercase `MOD`/`ADD`/`DEL` badge with a small colored dot indicator next to the filename. Icons stay consistent regardless of change state, exactly as the follow-up requested ("we DO NOT want to override/modify the icons themselves").

The new `buildFileChangeMap` / `buildChangedAncestors` / `classifyGitStatus` helpers translate git porcelain `XY` status codes into the existing `FileChangeType` union and are covered by their own unit-test file.

Closes #611.

## Test plan

- [x] `npx vitest run src/__tests__/renderer/utils/gitChangeMap.test.ts` (19 new unit tests pass)
- [x] `npx vitest run src/__tests__/renderer/components/FileExplorerPanel.test.tsx` (166 component tests pass, including new full-path / ancestor-walk / icon-consistency cases)
- [x] `npx tsc --noEmit -p tsconfig.app.json`
- [x] `npx tsc --noEmit -p tsconfig.test.json`
- [x] `npx prettier --check` on touched files
- [x] `npx eslint` on touched files (0 issues)
- [ ] Manually verify in dev build that modified / added / deleted indicators appear on the right files and that ancestor folders highlight

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * File explorer shows per-row git change dots (added/modified/deleted), folders marked when descendants changed, and colorblind-safe palette support.
  * Icons no longer get tinted by change state; row highlighting (bold + row color) applied for changed items.
* **Bug Fixes**
  * Change indicators require full relative path matches and gracefully handle missing/empty git data.
* **Tests**
  * Expanded tests covering change classification, mapping, ancestor detection, and UI indicators.

[Review Change Stack](https://app.coderabbit.ai/change-stack/RunMaestro/Maestro/pull/1007?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by pedramamini** [comment]
Thanks for the contribution, @chr1syy! Really nice fix.

The diagnosis is spot-on — `session.changedFiles` was indeed never being populated, and the `f.path.includes(node.name)` substring match was a latent bug waiting to bite anyone with shared leaf names. Wiring through `useGitDetail()` (which `GitStatusProvider` already polls per session) is the right move, and the VSCode-style ancestor walk plus the small colored dot indicator are exactly what the #611 follow-up asked for.

A few things I particularly liked:
- `gitChangeMap.ts` keeps the porcelain → `FileChangeType` translation in a pure, well-tested unit (the `AD` → `deleted` and `??` → `added` edge cases are nice).
- Icon set stays visually consistent — color signal lives on the filename + dot, per the original ask.
- Solid unit + component test coverage, including the explicit "siblings sharing a leaf name shouldn't false-match" regression test.

Approving. 🚀

File: src/renderer/utils/gitChangeMap.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ *
+ * Designed to be permissive: unknown codes fall through as `modified` so a
+ * dirty file is still surfaced rather than silently dropped.
+ */
+export function classifyGitStatus(status: string): FileChangeType {
+	if (status === '??') return 'added';
+	if (status.includes('D')) return 'deleted';
+	if (status.includes('A')) return 'added';
+	return 'modified';
+}
```
[the remark this answers is not in the document]
Comment (chr1syy) [reply src/renderer/utils/gitChangeMap.ts:24]: Addressed in 16476ae2d: tests now use the trimmed forms (`M`, `D`, `A`) that `useGitStatusPolling` actually stores, with an explicit case covering both trimmed and raw porcelain inputs. `classifyGitStatus` also trims defensively so callers don't have to remember which form they're holding.

File: src/renderer/components/FileExplorerPanel.tsx [hunk 2]
```diff
@@ -1172,44 +1195,41 @@ function FileExplorerPanelInner(props: FileExplorerPanelProps) {
...
-										: change.type === 'deleted'
-											? theme.colors.error
-											: theme.colors.warning,
+								backgroundColor: changeColor ?? theme.colors.textDim,
+								opacity: changeType ? 1 : 0.55,
 							}}
-						>
-							{change.type}
-						</span>
+						/>
```
[the remark this answers is not in the document]
Comment (chr1syy) [reply src/renderer/components/FileExplorerPanel.tsx:1232]: Addressed in 16476ae2d: the dot indicator now swaps to `COLORBLIND_STATUS_COLORS` (teal / orange / vermillion) when `colorBlindMode` is enabled, matching the same Wong-derived palette the default file icon already used. Added a component test that asserts the orange and teal hex values are applied for modified / added files in colorblind mode.
